File: frontend/src/utils/validation.ts
import { z } from 'zod';

export const textAnalysisSchema = z.object({
  text: z
    .string()
    .min(1, 'Text is required')
    .max(10000, 'Text must not exceed 10,000 characters'),
});

export const imageAnalysisSchema = z.object({
  file: z
    .instanceof(File)
    .refine((f) => f.size > 0, 'File is required')
    .refine(
      (f) => ['image/jpeg', 'image/png', 'image/webp', 'image/bmp'].includes(f.type),
      'File must be a supported image type (JPEG, PNG, WebP, BMP)'
    )
    .refine((f) => f.size <= 10 * 1024 * 1024, 'File must not exceed 10 MB'),
});

File: backend/core/config/validation.py
__all__ = [
    "MAX_TEXT_LENGTH", "MAX_FILE_SIZE_MB", "SUPPORTED_IMAGE_TYPES",
]

from typing import List

MAX_TEXT_LENGTH: int = 10000
MAX_FILE_SIZE_MB: int = 10
SUPPORTED_IMAGE_TYPES: List[str] = ["image/jpeg", "image/png", "image/webp", "image/bmp"]
